programs/exchange_api/src/exchange_processor.rs

```diff
@@ -738,7 +738,6 @@ mod test {
         );
 
         let instruction = ExchangeInstruction::new_swap_request(
-            &owner.pubkey(),
```

Comment:
@garious Here is an interesting case where the instruction does not need a signer but the transaction does.  The `client.process_instruction` call below panics because the owner is not part of the account keys.

Reply:
Neato!  So a signer to pay the transaction fee?

Reply:
Good point, I'll make the swap account the signer and payer of the fee

Reply:
So I'm working through git repo corruption problem so can't PR at the moment, but how about this? https://github.com/garious/solana/commit/11625e97038355891b29844530ec24d59acc8535

Reply:
Thanks @greg, I'm going to close this since the signer is needed anyway for the new account.